OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -28,9 +28,18 @@ public override Activity Tick(Actor self)
 			if (!target.IsValidFor(self))
 				return NextActivity;
 
-			var limitedAmmo = self.TraitOrDefault<LimitedAmmo>();
-			if (limitedAmmo != null && !limitedAmmo.HasAmmo())
-				return NextActivity;
+			var limitedAmmo = self.TraitsImplementing<LimitedAmmo>();
```

Comment:
> I'd rather add Actor self to the constructor and cache this there so we don't run it on every Tick.

OK. If I'm not mistaken, it might make sense to do the same for ```attack``` for the same reason, right?